statistikk/src/main/kotlin/no/nav/su/se/bakover/statistikk/stønad/StønadsstatistikkMapper.kt

```diff
@@ -0,0 +1,214 @@
+import no.nav.su.se.bakover.statistikk.ValidertStatistikkJsonMelding
+import org.slf4j.LoggerFactory
+import java.time.Clock
+import java.time.LocalDate
+import kotlin.math.roundToInt
+
+private val log = LoggerFactory.getLogger("StønadsstatistikkMapper.kt")
+
+private val stønadSchema: JsonSchema = SchemaValidator.createSchema("/statistikk/stonad_schema.json")
+
+internal fun StatistikkEvent.Stønadsvedtak.toStønadstatistikkDto(
+    aktørId: AktørId,
+    ytelseVirkningstidspunkt: LocalDate,
+    sak: Sak,
+    clock: Clock,
+    gitCommit: GitCommit?,
+): Either<Set<ValidationMessage>, ValidertStatistikkJsonMelding> {
+    return toDto(
+        vedtak = this.vedtak,
+        aktørId = aktørId,
+        ytelseVirkningstidspunkt = ytelseVirkningstidspunkt,
+        sak = sak,
+        clock = clock,
+        gitCommit = gitCommit,
+        funksjonellTid = this.vedtak.opprettet,
+    ).let {
+        serialize(it).let {
+            SchemaValidator.validate(it, stønadSchema).map {
+                ValidertStatistikkJsonMelding(
+                    topic = "supstonad.aapen-su-stonad-statistikk-v1",
+                    validertJsonMelding = it,
+                )
+            }
+        }
+    }
+}
+
+private fun toDto(
+    vedtak: VedtakSomKanRevurderes.EndringIYtelse,
+    aktørId: AktørId,
+    ytelseVirkningstidspunkt: LocalDate,
+    sak: Sak,
+    clock: Clock,
+    gitCommit: GitCommit?,
+    funksjonellTid: Tidspunkt,
+): StønadstatistikkDto {
+    return StønadstatistikkDto(
+        funksjonellTid = funksjonellTid,
+        tekniskTid = Tidspunkt.now(clock),
+        stonadstype = StønadstatistikkDto.Stønadstype.SU_UFØR,
+        sakId = vedtak.behandling.sakId,
+        aktorId = aktørId.toString().toLong(),
+        sakstype = vedtakstype(vedtak),
+        vedtaksdato = vedtak.opprettet.toLocalDate(zoneIdOslo),
+        vedtakstype = vedtakstype(vedtak),
+        vedtaksresultat = when (vedtak) {
+            is VedtakSomKanRevurderes.EndringIYtelse.GjenopptakAvYtelse -> StønadstatistikkDto.Vedtaksresultat.GJENOPPTATT
+            is VedtakSomKanRevurderes.EndringIYtelse.InnvilgetRevurdering -> StønadstatistikkDto.Vedtaksresultat.INNVILGET
+            is VedtakSomKanRevurderes.EndringIYtelse.InnvilgetSøknadsbehandling -> StønadstatistikkDto.Vedtaksresultat.INNVILGET
+            is VedtakSomKanRevurderes.EndringIYtelse.OpphørtRevurdering -> StønadstatistikkDto.Vedtaksresultat.OPPHØRT
+            is VedtakSomKanRevurderes.EndringIYtelse.StansAvYtelse -> StønadstatistikkDto.Vedtaksresultat.STANSET
+            is VedtakSomKanRevurderes.EndringIYtelse.InnvilgetRegulering -> StønadstatistikkDto.Vedtaksresultat.REGULERT
+        },
+        behandlendeEnhetKode = "4815",
+        ytelseVirkningstidspunkt = ytelseVirkningstidspunkt,
+        gjeldendeStonadVirkningstidspunkt = vedtak.behandling.periode.fraOgMed,
+        gjeldendeStonadStopptidspunkt = vedtak.behandling.periode.tilOgMed,
```

Comment:
Litt usikker på hva jeg ser på her, men det virker litt rart å sende med et "stopptidspunkt" for tilfeller hvor vi ikke har stanset ytelsen? Tilsvarende med utbetalingsstopp.

Reply:
Samme som masterkoden: https://github.com/navikt/su-se-bakover/blob/master/service/src/main/kotlin/no/nav/su/se/bakover/service/statistikk/mappers/St%C3%B8nadsstatistikkMapper.kt

Litt usikker selv. Fra doc: 
1. ytelseVirkningstidspunkt: Dato for når stønadsmottakers ytelse tredde i kraft første gang.
2. gjeldendeStonadVirkningstidspunkt: Dato for når gjeldende stønadsperiode startes.
3. gjeldendeStonadStopptidspunkt: Dato for når gjeldende stønadsperiode avsluttes.
4. gjeldendeStonadUtbetalingsstart: Dato for når utbetalingene starter for gjeldende stønadsperiode.
5. gjeldendeStonadUtbetalingsstopp: Dato for når utbetalingene stoppes for gjeldende stønadsperiode.

Reply:
Puttet det inn på trellokortet https://trello.com/c/IrzgBRnF